alodataset/coco_panoptic_dataset.py

```diff
@@ -0,0 +1,223 @@
+
+        # Create annotations
+        if "categories" in coco:
+            nb_category = max(cat["id"] for cat in coco["categories"])
+            self.label_names = ["N/A"] * (nb_category + 1)
+            for cat in coco["categories"]:
+                self.label_names[cat["id"]] = cat["name"]
+        print("Done")
+        return items
+
+    def get_split_ann_folder(self):
+        """Get annotation folder according to :attr:`split`.
+
+        Returns
+        -------
+        str
+            annotation folder
+        """
+        assert self.split in self.SPLIT_ANN_FOLDERS
+        return self.SPLIT_ANN_FOLDERS[self.split]
+
+    def get_split_ann_file(self):
+        """Get annotation file according to :attr:`split`.
+
+        Returns
+        -------
+        str
+            annotation file
+        """
+        assert self.split in self.SPLIT_ANN_FILES
+        return self.SPLIT_ANN_FILES[self.split]
+
+    def getitem(self, idx):
+        """Get the :mod:`Frame <aloscene.frame>` corresponds to *idx* index
+
+        Parameters
+        ----------
+        idx : int
+            Index of the frame to be returned
+
+        Returns
+        -------
+        :mod:`Frame <aloscene.frame>`
+            Frame with their corresponding boxes and masks attributes
+        """
+        if self.sample:
+            return BaseDataset.__getitem__(self, idx)
+
+        # Get elements
+        img_path, ann_path, ann_info = self.items[idx]
+
+        # Read mask annotation from panoptic image
+        masks = np.asarray(Image.open(ann_path), dtype=np.uint32)
+        masks = rgb2id(masks)  # Convert RGB to classesID
+
+        ids = np.array([ann["id"] for ann in ann_info["segments_info"]])
+        masks = masks == ids[:, None, None]
+        masks = torch.as_tensor(masks, dtype=torch.uint8)
+
+        labels = torch.as_tensor([ann["category_id"] for ann in ann_info["segments_info"]], dtype=torch.int64)
+
+        # Clean index by unique classes filtered
+        if self._ids_renamed is not None:
+            new_labels = self._ids_renamed[labels.numpy()]
+
+            # Keep only valid masks
+            idxs = np.where(new_labels >= 0)[0]
+            masks = masks[idxs]
+            labels = torch.from_numpy(new_labels[idxs])
+
+        # Make aloscene.frame
+        frame = Frame(img_path)
+
+        labels_2d = Labels(labels.to(torch.float32), labels_names=self.label_names, names=("N"), encoding="id")
+        boxes_2d = BoundingBoxes2D(
+            masks_to_boxes(masks),
+            boxes_format="xyxy",
+            absolute=True,
+            frame_size=frame.HW,
+            names=("N", None),
+            labels=labels_2d,
+        )
+        frame.append_boxes2d(boxes_2d)
+
+        if self.return_masks:
+            masks_2d = Mask(masks, names=("N", "H", "W"), labels=labels_2d)
+            frame.append_segmentation(masks_2d)
+        return frame
+
+
+if __name__ == "__main__":
+    coco_seg = CocoPanopticDataset()
```

Comment:
Should we upload the sample of `CocoPanopticDataset` ? and make this __main__ of this file used the sample instead of the real dataset ? (as all other datasets)

Reply:
Donde. For the moment `CocoPanopticDataset` uses the same `CocoDetectionDataset` sample. In #43 will be implement new labels for things/stuff classes. So the sample for `CocoPanopticDataset` will be update on that moment.